skbio/diversity/alpha/tests/test_base.py

```diff
@@ -45,6 +41,57 @@ def setUp(self):
                     u'0.75,(OTU5:0.25,(OTU6:0.5,OTU7:0.5):0.5):0.5):1.25):0.0'
                     u')root;'))
 
+    def test_alpha_diversity_invalid_input(self):
+        # number of ids doesn't match the number of samples
+        self.assertRaises(ValueError, alpha_diversity, 'observed_otus',
+                          self.table1, list('ABC'))
+
+        # otu_ids not provided for metric=faith_pd
+        self.assertRaises(ValueError, alpha_diversity, 'faith_pd', self.table1,
+                          list('ABC'), tree=self.t1)
+        # tree not provided for metric=faith_pd
+        self.assertRaises(ValueError, alpha_diversity, 'faith_pd', self.table1,
+                          list('ABC'), otu_ids=self.oids1)
+
+        # unknown metric provided
+        self.assertRaises(ValueError, alpha_diversity, 'not-a-metric',
+                          self.table1)
+
+    def test_alpha_diversity_observed_otus(self):
+        # expected values hand-calculated
+        expected = pd.Series([3, 3, 3, 3], index=self.sids1)
+        actual = alpha_diversity('observed_otus', self.table1, self.sids1)
+        npt.assert_array_equal(actual, expected)
+        # function passed instead of string
+        actual = alpha_diversity(observed_otus, self.table1, self.sids1)
+        npt.assert_array_equal(actual, expected)
+
+    def test_alpha_diversity_no_ids(self):
+        # expected values hand-calculated
+        expected = pd.Series([3, 3, 3, 3])
+        actual = alpha_diversity('observed_otus', self.table1)
+        npt.assert_array_equal(actual, expected)
+
+    def test_alpha_diversity_faith_pd(self):
+        # calling faith_pd through alpha_diversity gives same results as
+        # calling it directly
+        expected = []
+        for e in self.table1:
+            expected.append(faith_pd(e, tree=self.t1, otu_ids=self.oids1))
+        expected = pd.Series(expected)
+        actual = alpha_diversity('faith_pd', self.table1, tree=self.t1,
+                                 otu_ids=self.oids1)
+        npt.assert_array_equal(actual, expected)
```

Comment:
Use ``skbio.util._testing.assert_series_equal`` (need to define this based on [this](https://github.com/pydata/pandas/blob/master/pandas/util/testing.py#L867-L874)).